akka-actor-typed/src/main/scala/akka/actor/typed/ActorRefResolver.scala

```diff
@@ -49,8 +50,30 @@ abstract class ActorRefResolver extends Extension {
 
   private val untypedSystem = system.toUntyped.asInstanceOf[ExtendedActorSystem]
 
-  override def toSerializationFormat[T](ref: ActorRef[T]): String =
-    ref.path.toSerializationFormatWithAddress(untypedSystem.provider.getDefaultAddress)
+  override def toSerializationFormat[T](ref: ActorRef[T]): String = {
+
+    def toSerializationFormatWithAddress =
+      ref.path.toSerializationFormatWithAddress(untypedSystem.provider.getDefaultAddress)
+
+    ref.toUntyped match {
+      case a: ActorRefWithCell =>
+        val originSystem = a.underlying.system.asInstanceOf[ExtendedActorSystem]
+        if (originSystem eq untypedSystem)
+          toSerializationFormatWithAddress
+        else
+          throw new IllegalArgumentException(
+            s"ActorRefResolver for ActorSystem [${untypedSystem.provider.getDefaultAddress}] shouldn't be for " +
```

Comment:
```suggestion
            s"ActorRefResolver for ActorSystem [${untypedSystem.provider.getDefaultAddress}] shouldn't be used for " +
```

Reply:
pushed these on the side since no permission from ui

Reply:
weird :+1: